babel_datapipeline/util/misc.py

```diff
@@ -0,0 +1,16 @@
+def countnodes(infile):
+    distinct = set()
+    for line in infile:
+        paperIDs = line.split(' ')
```

Comment:
When using .split(' '), the tail will have a newline appended to it, which you want to strip. Otherwise PaperA PaperB will be extracted as PaperA PaperB\n.